vllm/worker/hpu_model_runner.py

```diff
@@ -1221,6 +1224,9 @@ def _prepare_decode(
         assert len(block_list) == len(block_groups)
         assert len(block_list) == len(block_usage)
 
+        if self.bucketing_ctx.num_hpu_blocks is None:
+            self.bucketing_ctx.num_hpu_blocks = self.cache_config.num_gpu_blocks
```

Comment:
Could we update the value right after profiling run instead of checking it in every decode run? Bucketing ctx is a singleton and should be accessible from worker.

Reply:
I was trying to set it in determine_num_available_blocks, but with CPU test this method was not being called.

Reply:
I think I see where the problem is